src/test/java/io/xstefank/wildfly/bot/PRReviewAssignmentTest.java

```diff
@@ -0,0 +1,149 @@
+import org.hamcrest.Matchers;
+import org.jboss.logmanager.Level;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.kohsuke.github.GHEvent;
+import org.kohsuke.github.GHPerson;
+import org.kohsuke.github.GHUser;
+import org.mockito.ArgumentCaptor;
+import org.mockito.ArgumentMatchers;
+import org.mockito.Mockito;
+
+import java.io.IOException;
+import java.util.List;
+import java.util.Set;
+import java.util.logging.LogManager;
+
+import static io.quarkiverse.githubapp.testing.GitHubAppTesting.given;
+import static io.xstefank.wildfly.bot.utils.TestConstants.TEST_REPO;
+
+@QuarkusTest
+@GitHubAppTest
+public class PRReviewAssignmentTest {
+
+    private static final java.util.logging.Logger rootLogger = LogManager.getLogManager().getLogger("io.xstefank.wildfly.bot");
+    private static final InMemoryLogHandler inMemoryLogHandler = new InMemoryLogHandler(
+            record -> record.getLevel().intValue() >= Level.ALL.intValue());
+
+    static {
+        rootLogger.addHandler(inMemoryLogHandler);
+    }
+
+    private static final String wildflyConfigFile = """
+                wildfly:
+                  rules:
+                    - id: "test"
+                      directories: [src]
+                      notify: [user1, user2]
+                  format:
+                    title:
+                      enabled: false
+                    commit:
+                      enabled: false
+                  emails:
+                    - foo@bar.baz
+                """;
+
+    private static GitHubJson gitHubJson;
+    private MockedGHPullRequestProcessor mockedContext;
+
+    @Inject
+    MockMailbox mailbox;
+
+    @BeforeAll
+    static void setupTests() throws IOException {
+        gitHubJson = GitHubJson.builder(TestConstants.VALID_PR_TEMPLATE_JSON).build();
+    }
+
+    @BeforeEach
+    void setup() {
+        mailbox.clear();
+    }
+
+    @Test
+    public void testOnlyMissingCollaboratorsNoReviewAssignment() throws IOException {
+        mockedContext = MockedGHPullRequestProcessor.builder(gitHubJson.id())
+                .prFile("src/main/java/resource/application.properties");
+        given().github(mocks -> Util.mockRepo(mocks, wildflyConfigFile, gitHubJson, mockedContext))
+                .when().payloadFromString(gitHubJson.jsonString())
+                .event(GHEvent.PULL_REQUEST)
+                .then().github(mocks -> {
+                    Mockito.verify(mocks.pullRequest(gitHubJson.id()), Mockito.never())
+                            .comment(ArgumentMatchers.anyString());
+                    Assertions.assertTrue(inMemoryLogHandler.getRecords().stream().anyMatch(
+                            logRecord -> logRecord.getMessage().equals(
+                                    "Following people are not collaborators in this repository [%s] and can not be requested for PR review: %s")));
```

Comment:
no formatting? there are %s expected?

Reply:
It extracts the logRecords, which do not have interpreted parameters yet